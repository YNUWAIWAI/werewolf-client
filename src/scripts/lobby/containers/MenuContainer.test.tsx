import * as ActionTypes from '../constants/ActionTypes'
import * as React from 'react'
import {IntlProviderContainer} from './IntlProviderContainer'
import {Menu} from '../components/organisms/Menu'
import {MenuContainer} from './MenuContainer'
import {Provider} from 'react-redux'
import fakeStore from './fakeStore'
import {mount} from 'enzyme'

describe('<MenuContainer />', () => {
  test('render', () => {
    const store = fakeStore()
    const wrapper = mount(
      <Provider store={store} >
        <IntlProviderContainer>
          <MenuContainer
            className="className"
            itemClassName="itemClassName"
            items={[]}
          />
        </IntlProviderContainer>
      </Provider>
    )

    expect(wrapper.html()).toMatchSnapshot()
  })
  test('transition', () => {
    const store = fakeStore()
    const dispatch = jest.fn()

    store.dispatch = dispatch
    const wrapper = mount(
      <Provider store={store} >
        <IntlProviderContainer>
          <MenuContainer
            className="className"
            itemClassName="itemClassName"
            items={[]}
          />
        </IntlProviderContainer>
      </Provider>
    )

    wrapper.find(Menu).props().transition(ActionTypes.AdvancedSearch.SEARCH)
    expect(dispatch).toHaveBeenCalledTimes(1)
    expect(dispatch).toHaveBeenCalledWith({
      type: ActionTypes.AdvancedSearch.SEARCH
    })
  })
})
