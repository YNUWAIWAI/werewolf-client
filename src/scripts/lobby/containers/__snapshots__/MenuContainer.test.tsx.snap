// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MenuContainer /> render 1`] = `"<ul class=\\"className\\"></ul>"`;
